store/store.go

```diff
@@ -608,6 +610,10 @@ type requestOptions struct {
 	ContentType  string
 	ExtraHeaders map[string]string
 	Data         []byte
+
+	// DeviceAuthIsForCustomStore indicates whether the operation strictly
+	// requires device auth only for custom stores to work.
+	DeviceAuthIsForCustomStore bool
```

Comment:
Sorry for the bikeshed, I find the name/comment a bit hard to grasp (maybe not enough tea yet?).

It seems like the idea behind this is that certain endpoints (sections, names) do not require device-auth unless there is a custom store involved. Maybe something like:
```
// SkipDeviceAuthForMainStore indicates that the given request does not need to perform device
// auth when talking to the main store. Custom stores always require device auth.
SkipDeviceAuthForMainStore bool
```

Reply:
I discovered that I need to generalize this in the next PR, I'll probably turn it into some kind of typed int already here